shell/run_cmd.go

```diff
@@ -69,6 +69,17 @@ func RunCommandWithOutput(
 			cmdStdout = io.MultiWriter(opts.Writer, &output.Stdout)
 		)
 
+		// Pass the traceparent to the child process if it is available in the context.
+		traceParent, err := telemetry.TraceParentFromContext(ctx)
+		if err != nil {
+			return errors.New(err)
+		}
+
+		if traceParent != "" {
+			opts.Logger.Debugf("Setting trace parent=%q for command %s", traceParent, fmt.Sprintf("%s %v", command, args))
+			opts.Env[telemetry.TraceParentEnv] = traceParent
```

Comment:
To confirm: We're supposed to throw away any `TRACEPARENT` environment variable value if something set it before calling Terragrunt? e.g. Jenkins

Reply:
Yes, I added test for this